lib/app/utils.js

```diff
@@ -69,9 +69,14 @@ export function getMatchedComponentsInstances(route, matches = false) {
 
 export function flatMapComponents(route, fn) {
   return Array.prototype.concat.apply([], route.matched.map(function (m, index) {
-    return Object.keys(m.components).map(function (key) {
-      return fn(m.components[key], m.instances[key], m, key, index)
-    })
+    return Object.keys(m.components).reduce(function (promises, key) {
```

Comment:
@clarkdo pardon my ignorance, but why do you use a full blown function in `reduce()` instead of `() => {}`? Is it because of previous compatibility with earlier Node versions?

Reply:
I guess since it's in `lib/app/utils.js` (and shipped to the browser), it's to avoid transpilation?

Reply:
@Atinux ha! Thought so! Thanks!